[CI] PackageTests test60Reinstall failing

I think this might be fallout from https://github.com/elastic/elasticsearch/pull/71505

**Build scan:**
https://gradle-enterprise.elastic.co/s/uht66ddl3ftic/tests/:qa:os:destructiveDistroTest.default-deb/org.elasticsearch.packaging.test.PackageTests/test60Reinstall

**Reproduction line:**
`null`

**Applicable branches:**
master, 7.x

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.packaging.test.PackageTests&tests.test=test60Reinstall

**Failure excerpt:**
```
org.elasticsearch.packaging.util.Shell$ShellException: Command was not successful: [bash -c sudo -u elasticsearch /usr/share/elasticsearch/bin/elasticsearch-keystore list]
   result: exitCode = [1] stdout = [] stderr = [Exception in thread "main" org.elasticsearch.common.settings.SettingsException: Failed to load settings from [elasticsearch.yml]
	at org.elasticsearch.common.settings.Settings$Builder.loadFromStream(Settings.java:1071)
	at org.elasticsearch.common.settings.Settings$Builder.loadFromPath(Settings.java:1044)
	at org.elasticsearch.node.InternalSettingsPreparer.prepareEnvironment(InternalSettingsPreparer.java:64)
	at org.elasticsearch.cli.EnvironmentAwareCommand.createEnv(EnvironmentAwareCommand.java:89)
	at org.elasticsearch.cli.EnvironmentAwareCommand.createEnv(EnvironmentAwareCommand.java:80)
	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:75)
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:116)
	at org.elasticsearch.cli.MultiCommand.execute(MultiCommand.java:80)
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:116)
	at org.elasticsearch.cli.Command.main(Command.java:79)
	at org.elasticsearch.common.settings.KeyStoreCli.main(KeyStoreCli.java:32)
Caused by: com.fasterxml.jackson.core.JsonParseException: Duplicate field 'ingest.geoip.downloader.enabled'
 at [Source: (sun.nio.ch.ChannelInputStream); line: 84, column: 32]
	at com.fasterxml.jackson.core.json.JsonReadContext._checkDup(JsonReadContext.java:204)
	at com.fasterxml.jackson.core.json.JsonReadContext.setCurrentName(JsonReadContext.java:198)
	at com.fasterxml.jackson.dataformat.yaml.YAMLParser.nextToken(YAMLParser.java:399)
	at org.elasticsearch.common.xcontent.json.JsonXContentParser.nextToken(JsonXContentParser.java:49)
	at org.elasticsearch.common.settings.Settings.fromXContent(Settings.java:603)
	at org.elasticsearch.common.settings.Settings.fromXContent(Settings.java:578)
	at org.elasticsearch.common.settings.Settings$Builder.loadFromStream(Settings.java:1067)
	... 10 more]

  at __randomizedtesting.SeedInfo.seed([30ABA2F80D8441D9:DA3B2FE65DAB1E48]:0)
  at org.elasticsearch.packaging.util.Shell.runScript(Shell.java:147)
  at org.elasticsearch.packaging.util.Shell.run(Shell.java:73)
  at org.elasticsearch.packaging.util.Packages.verifyOssInstallation(Packages.java:181)
  at org.elasticsearch.packaging.util.Packages.verifyPackageInstallation(Packages.java:152)
  at org.elasticsearch.packaging.test.PackageTests.test60Reinstall(PackageTests.java:219)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:566)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
  at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:834)

```

--- a/modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/DatabaseRegistry.java
+++ b/modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/DatabaseRegistry.java
@@ -349,7 +349,7 @@ public final class DatabaseRegistry implements Closeable {
                     // This makes the code easier to understand and maintain.
                     SearchResponse searchResponse = client.search(searchRequest).actionGet();
                     SearchHit[] hits = searchResponse.getHits().getHits();
-                    assert hits.length == 1 : "expected 1 hit, but instead got [" + hits.length + "]";
+
                     if (searchResponse.getHits().getHits().length == 0) {
                         failureHandler.accept(new ResourceNotFoundException("chunk document with id [" + id + "] not found"));
                         return;

--- a/qa/os/src/test/java/org/elasticsearch/packaging/util/ServerUtils.java
+++ b/qa/os/src/test/java/org/elasticsearch/packaging/util/ServerUtils.java
@@ -42,6 +42,7 @@ import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
 import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
 
 import static java.nio.file.StandardOpenOption.APPEND;
 import static java.nio.file.StandardOpenOption.CREATE;
@@ -253,6 +254,11 @@ public class ServerUtils {
 
     public static void disableGeoIpDownloader(Installation installation) throws IOException {
         List<String> yaml = Collections.singletonList("ingest.geoip.downloader.enabled: false");
-        Files.write(installation.config("elasticsearch.yml"), yaml, CREATE, APPEND);
+        Path yml = installation.config("elasticsearch.yml");
+        try (Stream<String> lines = Files.readAllLines(yml).stream()) {
+            if (lines.noneMatch(s -> s.startsWith("ingest.geoip.downloader.enabled"))) {
+                Files.write(installation.config("elasticsearch.yml"), yaml, CREATE, APPEND);
+            }
+        }
     }
 }

